examples/images_contours_and_fields/streamplot_demo.py

```diff
@@ -0,0 +1,82 @@
+"""
+==========
+Streamplot
+==========
+A streamplot, or streamline plot, is used to display 2D vector fields. This
+example shows a few features of the stream plot function:
+
+    * Varying the color along a streamline.
+    * Varying the density of streamlines.
+    * Varying the line width along a stream line.
+    * Controlling the start points of streamlines.
+    * Streamlines skipping masked regions and NaN values.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+
+w = 3
+Y, X = np.mgrid[-w:w:100j, -w:w:100j]
+U = -1 - X**2 + Y
+V = 1 + X - Y**2
+speed = np.sqrt(U*U + V*V)
+
+fig = plt.figure(figsize=(7, 9))
+gs = gridspec.GridSpec(nrows=3, ncols=2, height_ratios=[1, 1, 2])
+
+#  Varying density along a streamline
+ax0 = fig.add_subplot(gs[0, 0])
+ax0.streamplot(X, Y, U, V, density=[0.5, 1])
+ax0.set_title('Varying Density')
+
+# Varying color along a streamline
+ax1 = fig.add_subplot(gs[0, 1])
+strm = ax1.streamplot(X, Y, U, V, color=U, linewidth=2, cmap=plt.cm.autumn)
+fig.colorbar(strm.lines)
+ax1.set_title('Varying color')
+
+#  Varying line width along a streamline
+ax2 = fig.add_subplot(gs[1, 0])
+lw = 5*speed / speed.max()
+ax2.streamplot(X, Y, U, V, density=0.6, color='k', linewidth=lw)
+ax2.set_title('Varying Line Width')
+
+# Controlling the starting points of the streamlines
+X, Y = (np.linspace(-3, 3, 100),
+        np.linspace(-3, 3, 100))
+U, V = np.mgrid[-3:3:100j, 0:0:100j]
```

Comment:
Oh I think that I understand why the current situation was weird for both of us!

The current example in the official docs is indeed the one that you are refering to. However, the starting point example was changed in #6672 (by myself, which is why I was having the feeling that you changed U and V here) to use the same data as the other streamplot examples, which I think is *really* more demonstrative to show the different plotting options of `streamplot`. The related version of the example is only in our [devdocs gallery](http://matplotlib.org/devdocs/examples/images_contours_and_fields/streamplot_demo_start_points.html) (I guess because the PR was milestoned for 2.1 and not backported into 2.0). **So I would definitively support to use the same data (built once for all at the top of the script) for all the subplots :).**